drivers/timer/sam0_rtc_timer.c

```diff
@@ -23,7 +23,7 @@
 #define RTC0 ((RtcMode0 *) DT_RTC_SAM0_BASE_ADDRESS)
 
 /* Number of sys timer cycles per on tick. */
-#define CYCLES_PER_TICK (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC \
+#define CYCLES_PER_TICK (sys_clock_hw_cycles_per_sec() \
```

Comment:
This fails to build on SAM0, since the BUILD_ASSERT(s) below are no longer reference a constant.  

In this case it should likely be `SOC_ATMEL_SAM0_GCLK0_FREQ_HZ` since that's the actual source of the RTC clock, currently (it's also the source of the undivided CPU clock, so it works out either way at present).

Reply:
Fixed. Thank you. Please check if the fix is correct.

Reply:
Looks good to me now.